src/ceph.in

```diff
@@ -552,8 +552,8 @@ def ping_monitor(cluster_handle, name, timeout):
         return 1
 
     mon_id = name[len('mon.'):]
+    run_in_thread(cluster_handle.connect, timeout=timeout)
```

Comment:
> kefu: not in the form of a [mon] declaration, no.   to be specific, I have:
> mon_initial_members = mira021, mira049, mira060
> mon_host = 172.21.4.136, 172.21.5.114, 172.21.6.130

if you are using `ceph ping mon.a`, the client will try to ping `172.21.4.136`, if `mon.b`, client will ping `172.21.5.114`.

> > afaict, mon_host is the only relevant thing to build_initial_monmap

and `build_initial_monmap()` enumerate all ceph.conf sections starts with `"mon."` also if your `"mon_host"` is empty or not set.

> I have a ceph.conf with mon_initial_members and mon_host set, but no monmap; this causes build_initial_monmap to return a monmap with 'noname-a', 'noname-b', etc. set in it; then, MonClient::ping_monitor tries to find a monitor named 'noname-<specific>', which of course fails and returns ENOENT.

it returns `ENOENT` only there is no such a monitor named 'noname-<specific>', but `MonClient::ping_monitor()` won't search for "noname-${mon_id}" unless `monmap.contains("noname-"+mon_id)`, see https://github.com/ceph/ceph/blob/master/src/mon/MonClient.cc#L205. otherwise it will just try to find "${specific}" in the monmap.